test/s3path.jl

```diff
@@ -334,4 +335,33 @@ end
     end
 end
 
+@testset "S3Path versioning" begin
+    key_version_file = "test_versions"
+    s3_put(bucket_name, key_version_file, "data.v1")
+    s3_put(bucket_name, key_version_file, "data.v2")
+
+    versions = s3_list_versions(aws, bucket_name, key_version_file)
+    @test length(versions) == 2 # versions is ordered newest to oldest---`first(versions)` is the latest version
```

Comment:
Probably more intuitive to do:
```suggestion
    # `s3_list_versions` returns versions in the order newest to oldest
    versions = reverse!(s3_list_versions(aws, bucket_name, key_version_file))
    @test length(versions) == 2 
```
This will require additional changes below.